app/connectors/cache/MongoCacheConnector.scala

```diff
@@ -26,55 +27,114 @@ import uk.gov.hmrc.play.frontend.auth.AuthContext
 
 import scala.concurrent.Future
 
-class MongoCacheConnector @Inject()(cacheClientFactory: MongoCacheClientFactory) extends Conversions {
+class MongoCacheConnector @Inject()(cacheClientFactory: MongoCacheClientFactory, authConnector: AuthConnector) extends Conversions {
 
   lazy val mongoCache: MongoCacheClient = cacheClientFactory.createClient
 
   /**
     * Fetches the data item with the specified key from the mongo store
     */
-  def fetch[T](key: String)(implicit authContext: AuthContext, hc: HeaderCarrier, formats: Format[T]): Future[Option[T]] =
+  def fetchByOid[T](key: String)(implicit authContext: AuthContext, hc: HeaderCarrier, formats: Format[T]): Future[Option[T]] =
     mongoCache.find(authContext.user.oid, key)
 
+  def fetchByCredId[T](key: String)(implicit authContext: AuthContext, hc: HeaderCarrier, formats: Format[T]): Future[Option[T]] = {
+    authConnector.getCredId flatMap {
+      credId => mongoCache.findWithCacheMiss(authContext.user.oid, credId, key)
+    }
+  }
+
+  def fetch[T](key: String)(implicit authContext: AuthContext, hc: HeaderCarrier, formats: Format[T]): Future[Option[T]] = {
+    fetchByCredId(key) flatMap {
+      case Some(data) => Future.successful(Some(data))
+      case _ => fetchByOid(key)
+    }
+  }
+
   /**
     * Saves the data item in the mongo store with the specified key
     */
-  def save[T](key: String, data: T)(implicit authContext: AuthContext, hc: HeaderCarrier, format: Format[T]): Future[CacheMap] =
-    mongoCache.createOrUpdate(authContext.user.oid, data, key).map(toCacheMap)
+  def save[T](key: String, data: T)(implicit authContext: AuthContext, hc: HeaderCarrier, format: Format[T]): Future[CacheMap] = {
+    authConnector.getCredId flatMap {
+      credId => {
+        mongoCache.createOrUpdateWithCacheMiss(authContext.user.oid, credId, data, key).map(toCacheMap)
```

Comment:
I would rather have credId passed in before oid

Reply:
Because?

Reply:
ok, will do

Reply:
done